mlos_bench/mlos_bench/environments/local/local_fileshare_env.py

```diff
@@ -140,15 +137,11 @@ def run(self) -> Tuple[Status, Optional[dict]]:
             If run script is a benchmark, then the score is usually expected to
             be in the `score` field.
         """
-        prev_temp_dir = self._temp_dir
-        with self._local_exec_service.temp_dir_context(self._temp_dir) as self._temp_dir:
-            # Override _temp_dir so that download and run both use the same path.
-            params = self._get_env_params()
-            params["PWD"] = self._temp_dir
-            for (path_from, path_to) in self._expand(self._download, params):
-                self._file_share_service.download(
-                    path_from, self._config_loader_service.resolve_path(
-                        path_to, extra_paths=[self._temp_dir]))
-            result = super().run()
-            self._temp_dir = prev_temp_dir
-            return result
+        assert self._temp_dir is not None
+        params = self._get_env_params()
+        params["PWD"] = self._temp_dir
+        for (path_from, path_to) in self._expand(self._download, params):
+            self._file_share_service.download(
+                path_from, self._config_loader_service.resolve_path(
+                    path_to, extra_paths=[self._temp_dir]))
+        return super().run()
```

Comment:
Order of this seems a little funny to me.  The assert (not enabled yet) would happen after the work is being done, and it would return some results from the parent rather than the target callee?

Reply:
yes, the parent of `LocalFileShareEnv` is `LocalEnv`, and it has logic that has to be executed _after_ the download. The assertion part is a bit wacky, but we have the assertions in `.setup()` already.. Anyway, I think it won't hurt to add one more assertion before the download

Reply:
oh wait, we do have an assert at teh beginning of `LocalFileShareEnv.run()`!